css-scroll-snap-2/Overview.bs

```diff
@@ -470,3 +480,114 @@ Physical Longhands for 'scroll-start-target' {#scroll-start-target-longhands-phy
 	</pre>
 
 	...
+
+Appendix B: Snap Events Timing {#snap-events-timing}
+==================================
+
+This appendix describes the algorithm UAs should follow to dispatch {{snapchanged}}
+and {{snapchanging}} events in response to a scrolling operation on a
+<a spec=css-scroll-snap lt="scroll snap container">snap container</a>.
+
+Each {{Document}} has an associated list of
+<dfn export for=Document>pending snapchanging event targets</dfn>, initially
+empty.
+
+Each {{Document}} associates each of its
+<a spec=css-scroll-snap lt="scroll snap container">snap containers</a>
+with at most one <dfn>snapchangingTargetBlock</dfn> and at most one
+<dfn>snapchangingTargetInline</dfn> in the block and inline axes respectively.
+
+When a <a spec=css-scroll-snap lt="scroll snap container">snap container</a> is
+scrolled, run these steps:
+
+0. Let <var>doc</var> be the
+	 <a spec=css-scroll-snap lt="scroll snap container">snap container</a>'s
+	 associated {{Document}}.
+1. Let <var>scrollPosition</var> be the scroll offset resulting from applying
+	 all pending scroll updates to the snap container.
+2. Let <var>blockTarget</var> be the
+	 <a>snapchangingTargetBlock</a> that <var>doc</var> associates with the snap
+	 container.
+3. Let <var>inlineTarget</var> be the <a>snapchangingTargetInline</a> that
+	 <var>doc</var> associates with the snap container.
+4. Let <var>newBlockTarget</var> be the element that the UA would snap the snap
+	 container to from <var>scrollPosition</var> (or null if it would not snap to
+	 any element) along the block axis.
+5. Let <var>newInlineTarget</var> be the element that the UA would snap the snap
+	 container to from <var>scrollPosition</var> (or null if it would not snap to
+	 any element) along the inline axis.
+6. If <var>newBlockTarget</var> is not the same element as
+	 <var>blockTarget</var>:
+	
+	1. Set <var>doc</var>'s <a>snapchangingTargetBlock</a> for the snap container
+		 to <var>newBlockTarget</var>.
+	2. Append the snap container to <var>doc</var>'s
+		 <a>pending snapchanging event targets</a> 
+7. If <var>newInlineTarget</var> is not the same element as
+	 <var>inlineTarget</var>:
+
+	1. Set <var>doc</var>'s <a>snapchangingTargetInline</a> for the snap container
+		 to <var>newInlineTarget</var>.
+	2. Append the snap container to <var>doc</var>'s pending snapchanging event
+		 targets.
```

Comment:
I think you only want to append the snap container to the pending snapchanging event targets list once. Note that cssom-view says to only do this if the target is not already in the pending scroll event targets.

Reply:
Yes, I've added wording to make it clear that the snap container should only be added to the list if it's not already there.